lib/board.rb

```diff
@@ -0,0 +1,69 @@
+class Board
+
+  attr_accessor :cells
+
+
+  def initialize
+    self.cells ||= Array.new(9, " ")
+  end # initialize
+
+
+  ## displays the board
+  def display
+
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+
+  end # board
+
+
+  ## resets the array if necessary
+  def reset!
+    self.cells = Array.new(9, " ")
+  end # reset!
+
+
+  ## determines the value in the input position
+  def position(user_input)
+    cells[user_input.to_i - 1]
+  end # position
+
+
+  ## determines if the board is full
+  def full?
+    true if cells.all?{|pos| pos == "X" || pos == "O"}
+  end # full?
+
+
+  ## determines what turn it is now
+  def turn_count
+    count = 0
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```